cmd/src/codeowners_delete.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+
+	"github.com/sourcegraph/src-cli/internal/api"
+	"github.com/sourcegraph/src-cli/internal/cmderrors"
+)
+
+func init() {
+	usage := `
+Examples:
+
+  Delete a codeowners file for the repository "github.com/sourcegraph/sourcegraph":
+
+    	$ src codeowners delete -repo='github.com/sourcegraph/sourcegraph'
+`
+
+	flagSet := flag.NewFlagSet("delete", flag.ExitOnError)
+	usageFunc := func() {
+		fmt.Fprintf(flag.CommandLine.Output(), "Usage of 'src codeowners %s':\n", flagSet.Name())
+		flagSet.PrintDefaults()
+		fmt.Println(usage)
+	}
+	var (
+		repoFlag = flagSet.String("repo", "", "The repository to delete the data for")
+		apiFlags = api.NewFlags(flagSet)
+	)
+
+	handler := func(args []string) error {
+		if err := flagSet.Parse(args); err != nil {
+			return err
+		}
+
+		if *repoFlag == "" {
+			return errors.New("provide a repo name")
+		}
+
+		client := cfg.apiClient(apiFlags, flagSet.Output())
+
+		query := `mutation DeleteCodeownersFile(
+	$repoName: String!,
+) {
+	deleteCodeownersFiles(repositories: [{
+		repoName: $repoName,
+	}]) {
+		alwaysNil
+	}
+}
```

Comment:
I think this isn't up to date with main anymore, it takes a slice (just in case we wanted an admin UI with it)

```
go run ./cmd/src codeowners delete -repo 'github.com/leonore/sourcegraph-test'
GraphQL errors: 2 errors occurred:
	* {
  "locations": [
    {
      "column": 24,
      "line": 4
    }
  ],
  "message": "Unknown argument \"repositories\" on field \"deleteCodeownersFiles\" of type \"Mutation\"."
}
	* {
  "locations": [
    {
      "column": 2,
      "line": 4
    }
  ],
  "message": "Field \"deleteCodeownersFiles\" argument \"repositoryIDs\" of type \"[Int!]!\" is required but not provided."
}
exit status 1
```

Reply:
did you try it as of https://github.com/sourcegraph/sourcegraph/pull/48584?